Release Packages for Questing

frenchwr opened the pull request:
The packages are published in the [oneapi-release PPA here](https://launchpad.net/~kobuk-team/+archive/ubuntu/oneapi-release?field.series_filter=questing).

I have also validated the packages on `mustang` (which contains a Battlemage GPU) running 25.10 using the autopkgtests for the compiler [1] and oneDPL [2] libraries.

There are also a few changes to the dependencies worth noting:

* At run time, I am now installing Intel graphics packages from the Ubuntu archive rather than copying over packages from the `intel-graphics` PPA, which was my previous approach for plucky
* At build time, I am now using `libparallel-hashmap-dev` from the Ubuntu archive (which is synced from debian). The version from the archive is 1.4.1, which is a version behind the version used in Intel's build (2.0.0). Checking the [release notes](https://github.com/greg7mdp/parallel-hashmap/releases/tag/v2.0.0), the changes in 2.0.0 are fairly minimal ([link to the diff](https://github.com/greg7mdp/parallel-hashmap/compare/v1.4.1...v2.0.0)). Specifically, the compiler only uses one header from parallel-hashmap, `phmap.h` ([link](https://github.com/intel/llvm/blob/v6.2.0/xptifw/src/xpti_trace_framework.cpp#L17)). That header had some changes between 1.4.1 and 2.0.0, but they were almost entirely related to the introduction of a `PHMAP_DISABLE_MIX` macro that is not relevant to the DPC++ compiler. Additionally, the fact that the compiler built successfully with version 1.4.1, and that our autopkgtests passed, suggests that there is not a strong dependency on the version. I suspect that Intel is simply bumping to the latest version available at the time of the compiler build/release. This is great because we avoid needing to bump the version in debian or Ubuntu.

[1] 

```
$ autopkgtest -B -- null
autopkgtest [13:20:02]: starting date and time: 2025-10-17 13:20:02+0000
autopkgtest [13:20:02]: version 5.49
autopkgtest [13:20:02]: host mustang-118102; command line: /usr/bin/autopkgtest -B -- null
autopkgtest [13:20:02]: testbed dpkg architecture: amd64
autopkgtest [13:20:02]: testbed apt version: 3.1.6ubuntu2
autopkgtest [13:20:02]: @@@@@@@@@@@@@@@@@@@@ test bed setup
autopkgtest [13:20:02]: testbed release detected to be: None
autopkgtest [13:20:03]: testbed running kernel: Linux 6.17.0-5-generic #5-Ubuntu SMP PREEMPT_DYNAMIC Mon Sep 22 10:00:33 UTC 2025
autopkgtest [13:20:03]: @@@@@@@@@@@@@@@@@@@@ unbuilt-tree .
autopkgtest [13:20:03]: testing package intel-dpcpp version 6.2.0-0ubuntu1~25.10~ppa8
autopkgtest [13:20:03]: build not needed
autopkgtest [13:20:03]: test sycl-neo-device: preparing testbed
autopkgtest [13:20:03]: test sycl-neo-device: [-----------------------
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/sycl-neo-device] Starting build phase
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/sycl-neo-device] Starting run phase
No device of requested type available.
autopkgtest [13:20:06]: test sycl-neo-device: -----------------------]
autopkgtest [13:20:06]: test sycl-neo-device:  - - - - - - - - - - results - - - - - - - - - -
sycl-neo-device      PASS
autopkgtest [13:20:06]: test mandelbrot: preparing testbed
autopkgtest [13:20:06]: test mandelbrot: [-----------------------
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/mandelbrot] Starting build phase
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Default CMAKE_BUILD_TYPE not set using Release with Debug Info
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is Clang 21.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.5s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/002-mandelbrot
[ 25%] Building CXX object src/CMakeFiles/mandelbrot.dir/main.cpp.o
[ 50%] Linking CXX executable mandelbrot
[ 50%] Built target mandelbrot
[ 75%] Building CXX object src/CMakeFiles/mandelbrot_usm.dir/main.cpp.o
[100%] Linking CXX executable mandelbrot_usm
[100%] Built target mandelbrot_usm
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/mandelbrot] Standard run
     Platform Name: Intel(R) oneAPI Unified Runtime over Level-Zero
  Platform Version: 1.6
       Device Name: Intel(R) Arc(TM) B580 Graphics
    Max Work Group: 1024
 Max Compute Units: 160

Parallel Mandelbrot set using buffers.
 Rendered image output to file: mandelbrot.png (output too large to display in text)
       Serial time: 0.0260966s
     Parallel time: 0.00934625s
Successfully computed Mandelbrot set.
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/mandelbrot] Unified shared memory (USM) run
     Platform Name: Intel(R) oneAPI Unified Runtime over Level-Zero
  Platform Version: 1.6
       Device Name: Intel(R) Arc(TM) B580 Graphics
    Max Work Group: 1024
 Max Compute Units: 160

Parallel Mandelbrot set using USM.
 Rendered image output to file: mandelbrot.png (output too large to display in text)
       Serial time: 0.0262101s
     Parallel time: 3.85218e-05s
Successfully computed Mandelbrot set.
autopkgtest [13:20:19]: test mandelbrot: -----------------------]
autopkgtest [13:20:19]: test mandelbrot:  - - - - - - - - - - results - - - - - - - - - -
mandelbrot           PASS
autopkgtest [13:20:19]: test vector-add: preparing testbed
autopkgtest [13:20:19]: test vector-add: [-----------------------
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/vector-add] Starting build phase - buffer implementation
CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 21.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/003-vector-add/buffers
[ 50%] Building CXX object src/CMakeFiles/vector-add-buffers.dir/vector-add-buffers.cpp.o
[100%] Linking CXX executable ../vector-add-buffers
[100%] Built target vector-add-buffers
[100%] Built target cpu-gpu
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/vector-add] Starting run phase - buffer implementation
Running on device: Intel(R) Arc(TM) B580 Graphics
Vector size: 10000
[0]: 0 + 0 = 0
[1]: 1 + 1 = 2
[2]: 2 + 2 = 4
...
[9999]: 9999 + 9999 = 19998
Vector add successfully completed on device.
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/vector-add] Starting build phase - USM implementation
CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 21.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using the USM variant.
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/003-vector-add/usm
[ 50%] Building CXX object src/CMakeFiles/vector-add-usm.dir/vector-add-usm.cpp.o
[100%] Linking CXX executable ../vector-add-usm
[100%] Built target vector-add-usm
[100%] Built target cpu-gpu
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/vector-add] Starting run phase - USM implementation
Running on device: Intel(R) Arc(TM) B580 Graphics
Vector size: 10000
[0]: 0 + 0 = 0
[1]: 1 + 1 = 2
[2]: 2 + 2 = 4
...
[9999]: 9999 + 9999 = 19998
Vector add successfully completed on device.
autopkgtest [13:20:27]: test vector-add: -----------------------]
autopkgtest [13:20:27]: test vector-add:  - - - - - - - - - - results - - - - - - - - - -
vector-add           PASS
autopkgtest [13:20:27]: test matrix-mul: preparing testbed
autopkgtest [13:20:27]: test matrix-mul: [-----------------------
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/matrix-mul] Starting build phase
clang++-dpcpp  -std=c++17 -fsycl -g -o matrix_mul_dpc src/matrix_mul_sycl.cpp 
>>> [/tmp/autopkgtest.tMtYOs/build.24o/real-tree/debian/tests/matrix-mul] Starting run phase
Device: Intel(R) Arc(TM) B580 Graphics
Problem size: c(150,600) = a(150,300) * b(300,600)
Result of matrix multiplication using SYCL: Success - The results are correct!
autopkgtest [13:20:31]: test matrix-mul: -----------------------]
autopkgtest [13:20:31]: test matrix-mul:  - - - - - - - - - - results - - - - - - - - - -
matrix-mul           PASS
autopkgtest [13:20:31]: @@@@@@@@@@@@@@@@@@@@ summary
sycl-neo-device      PASS
mandelbrot           PASS
vector-add           PASS
matrix-mul           PASS
```

[2]

```
$ autopkgtest -B -- null
autopkgtest [13:27:44]: starting date and time: 2025-10-17 13:27:44+0000
autopkgtest [13:27:44]: version 5.49
autopkgtest [13:27:44]: host mustang-118102; command line: /usr/bin/autopkgtest -B -- null
autopkgtest [13:27:44]: testbed dpkg architecture: amd64
autopkgtest [13:27:44]: testbed apt version: 3.1.6ubuntu2
autopkgtest [13:27:44]: @@@@@@@@@@@@@@@@@@@@ test bed setup
autopkgtest [13:27:44]: testbed release detected to be: None
autopkgtest [13:27:44]: testbed running kernel: Linux 6.17.0-5-generic #5-Ubuntu SMP PREEMPT_DYNAMIC Mon Sep 22 10:00:33 UTC 2025
autopkgtest [13:27:44]: @@@@@@@@@@@@@@@@@@@@ unbuilt-tree .
autopkgtest [13:27:44]: testing package onedpl version 2022.9.0-0ubuntu1~25.10~ppa8
autopkgtest [13:27:44]: build not needed
autopkgtest [13:27:44]: test reduce-segment: preparing testbed
autopkgtest [13:27:44]: test reduce-segment: [-----------------------
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/reduce-segment] Starting build phase
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/reduce-segment] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
Keys = [ 0 0 0 1 1 1 ]
Values = [ 1 2 3 4 5 6 ]
Output Keys = [ 0 1 ]
Output Values = [ 6 15 ]
autopkgtest [13:27:51]: test reduce-segment: -----------------------]
autopkgtest [13:27:51]: test reduce-segment:  - - - - - - - - - - results - - - - - - - - - -
reduce-segment       PASS
autopkgtest [13:27:51]: test binary-search: preparing testbed
autopkgtest [13:27:51]: test binary-search: [-----------------------
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/binary-search] Starting build phase
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/binary-search] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
Input sequence = [0 2 2 2 3 3 3 3 6 6 ]
Search sequence = [0 2 4 7 6 ]
Search results = [1 1 0 0 1 ]
autopkgtest [13:27:56]: test binary-search: -----------------------]
autopkgtest [13:27:56]: test binary-search:  - - - - - - - - - - results - - - - - - - - - -
binary-search        PASS
autopkgtest [13:27:56]: test dpl-sortdouble: preparing testbed
autopkgtest [13:27:56]: test dpl-sortdouble: [-----------------------
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/dpl-sortdouble] Starting build phase
>>> [/tmp/autopkgtest.Bvjx7a/build.aNv/real-tree/debian/tests/dpl-sortdouble] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
2
4
6
8
autopkgtest [13:28:04]: test dpl-sortdouble: -----------------------]
autopkgtest [13:28:04]: test dpl-sortdouble:  - - - - - - - - - - results - - - - - - - - - -
dpl-sortdouble       PASS
autopkgtest [13:28:04]: @@@@@@@@@@@@@@@@@@@@ summary
reduce-segment       PASS
binary-search        PASS
dpl-sortdouble       PASS
```

frenchwr committed:
Remove validation dir in favor of autopkgtests

frenchwr committed:
Remove vc-intrinsics.

This package is now being used in-tree for the compiler build.

frenchwr committed:
Build onedpl for questing

frenchwr committed:
Build intel-umf for questing

frenchwr committed:
Build emhash for questing

frenchwr committed:
Build compiler for questing

frenchwr committed:
Update README for Questing

frenchwr commented:
About parallel-hashmap, I opened a GitHub issue with the DPC++ compiler team to ask about the version: https://github.com/intel/llvm/issues/20406

mckees commented on onedpl/README.md:
````diff
@@ -6,13 +6,55 @@ To update and build:
-sbuild -c plucky-amd64 --dist=plucky --build-path=""
+```
+
+If this is a new upstream version or you are building for a new series for the first time, you need to rename the .tar.xz source file and delete the sym link, as the sym link is not handled properly for `dpkg-buildpackage` and `dput` (TODO: research how to overcome this manual step). For example:
````
Shouldn't this be fixed before we upload to the archive?

frenchwr replied:
Done!

frenchwr committed:
Modify d/watch files, cleanup d/changelog.

Update d/watch files to generate an orig tarball without
using sym links, and remove old changelogs entries that
were used during package development.

frenchwr committed:
Update PPA name, add missing dep

frenchwr commented:
@mckees This is ready for review again. I have everything published to this new PPA: https://launchpad.net/~kobuk-team/+archive/ubuntu/oneapi-release

mckees commented:
LGTM, thanks!
